recipes/libavrocpp/all/conanfile.py

```diff
@@ -42,10 +43,13 @@ def configure(self):
 
     def layout(self):
         cmake_layout(self, src_folder="src")
-
+        
     def requirements(self):
-        self.requires("boost/1.81.0")
-        self.requires("snappy/1.1.9")
+        if valid_min_cppstd(self, 14) :
+            self.requires("boost/[>=1.82.0]", transitive_headers=True)
+        else:
+            self.requires("boost/[<1.82.0]", transitive_headers=True)
+        self.requires("snappy/[~1.1.9]")
```

Comment:
Unfortunately we'll need to keep these requirements pinned to a specific version -while we have added some libraries to the list of allowed recipes, boost/snappy are not part of those for now :/